components/tidb_query/src/codec/chunk/chunk.rs

```diff
@@ -56,6 +59,83 @@ impl Chunk {
         self.columns[col_idx].append_datum(v)
     }
 
+    /// Append a datum from vec to the column
+    #[inline]
+    pub fn append_vec(
```

Comment:
You can append a whole vector to the column, which will save a lot of unnecessary type matches.